utils/src/test/java/org/robolectric/util/PerfStatsCollectorTest.java

```diff
@@ -47,21 +52,81 @@ public void shouldMeasureElapsedTimeForRepeatedEvents() throws Exception {
 
     Collection<Metric> metrics = collector.getMetrics();
     assertThat(metrics).containsExactly(
-        new Metric("repeatable event", 3, 60)
+        new Metric("repeatable event", 3, 60, true)
     );
   }
 
+  @Test
+  public void shouldRunAndMeasureSuccessfulCallable() throws Exception {
+    assertThat(collector.measure("event", () -> {
+      fakeClock.delay(10);
+      return "return value";
+    })).isEqualTo("return value");
+
+    Collection<Metric> metrics = collector.getMetrics();
+    assertThat(metrics).containsExactly(new Metric("event", 1, 10, true));
+  }
+
+  @Test
+  public void shouldRunAndMeasureExceptionThrowingCallable() throws Exception {
+    collector.measure("event", () -> {
+      fakeClock.delay(10);
+      return "return value";
+    });
+
+    try {
+      collector.measure("event", () -> {
+        fakeClock.delay(5);
+        throw new RuntimeException("fake");
+      });
+      fail("should have thrown");
+    } catch (RuntimeException e) {
+      assertThat(e.getMessage()).isEqualTo("fake");
+    }
+
+    Collection<Metric> metrics = collector.getMetrics();
+    assertThat(metrics).containsExactlyInAnyOrder(
+        new Metric("event", 1, 10, true),
+        new Metric("event", 1, 5, false));
+  }
+
+  @Test
+  public void shouldRunAndMeasureCheckedException() throws Exception {
+    try {
+      collector.measure("event", () -> {
+        fakeClock.delay(5);
+        throw new IOException("fake");
+      });
+      fail("should have thrown");
+    } catch (IOException e) {
+      assertThat(e.getMessage()).isEqualTo("fake");
+    }
+
+    Collection<Metric> metrics = collector.getMetrics();
+    assertThat(metrics).containsExactlyInAnyOrder(
+        new Metric("event", 1, 5, false));
+  }
+
+  @Test
+  public void reset_shouldClearAllMetadataAndMetrics() throws Exception {
+    collector.putMetadata(String.class, "metadata");
+    collector.startEvent("event").finished();
+    collector.reset();
+    assertThat(collector.getMetadata().get(String.class)).isNull();
+    assertThat(collector.getMetrics()).isEmpty();
+  }
+
   private class FakeClock implements Clock {
 
-    private int timeMs = 0;
+    private int timeNs = 0;
 
     @Override
-    public long currentTimeMillis() {
-      return timeMs;
+    public long nanoTime() {
```

Comment:
I don't feel this name is meaningful enough. What about `currentTimeNanos()`?

Reply:
I think Java picked [`nanoTime()`](https://docs.oracle.com/javase/7/docs/api/java/lang/System.html#nanoTime()) over something like `currentTimeNanos()` because, unlike `currentTimeMillis()`, the nanosecond-resolution counter isn't relative to a fixed universal point in time, it's only 'current' delta itself. It's impossible to convert from nanos to a calendar date and time.